src/consensus/pbft/libbyz/Replica.cpp

```diff
@@ -343,31 +334,104 @@ bool Replica::compare_execution_results(
              << ", info.ctx:" << info.ctx << std::endl;
     return false;
   }
+
+  if (!std::equal(
+        std::begin(pp_root),
+        std::end(pp_root),
+        std::begin(info.full_state_merkle_root)))
+  {
+    LOG_FAIL << "Full state merkle root between execution and the pre_prepare "
+                "message does not match, seqno:"
+             << pre_prepare->seqno() << std::endl;
+    return false;
+  }
+
   return true;
 }
 
-void Replica::playback_request(const pbft::Request& request)
+void Replica::playback_transaction(ccf::Store::Tx& tx)
 {
+  if (vtimer->get_state() == ITimer::State::running)
+  {
+    vtimer->restart();
+  }
+  {
```

Comment:
need to check if has writes since we store all requests and pre-prepares at `key 0` so might be populated but not fresh